Sources/AsyncHTTPClient/HTTPClient+HTTPCookie.swift

```diff
@@ -42,79 +47,71 @@ extension HTTPClient {
         public init?(header: String, defaultDomain: String) {
-            let components = header.components(separatedBy: ";").map {
-                $0.trimmingCharacters(in: .whitespaces)
-            }
-
-            if components.isEmpty {
+            // The parsing of "Set-Cookie" headers is defined by Section 5.2, RFC-6265:
+            // https://datatracker.ietf.org/doc/html/rfc6265#section-5.2
+            var components = header.utf8.split(separator: UInt8(ascii: ";"), omittingEmptySubsequences: false)[...]
+            guard let keyValuePair = components.popFirst()?.trimmingASCIISpaces() else {
                 return nil
             }
-
-            let nameAndValue = components[0].split(separator: "=", maxSplits: 1, omittingEmptySubsequences: false).map {
-                $0.trimmingCharacters(in: .whitespaces)
-            }
-
-            guard nameAndValue.count == 2 else {
+            guard let (trimmedName, trimmedValue) = keyValuePair.parseKeyValuePair() else {
                 return nil
             }
-
-            self.name = nameAndValue[0]
-            self.value = nameAndValue[1].omittingQuotes()
-
-            guard !self.name.isEmpty else {
+            guard !trimmedName.isEmpty else {
                 return nil
             }
 
-            self.path = "/"
-            self.domain = defaultDomain
-            self.expires = nil
+            self.name = String(aligningUTF8: trimmedName)
+            self.value = String(aligningUTF8: trimmedValue.trimmingPairedASCIIQuote())
+            self.expires_timestamp = nil
             self.maxAge = nil
             self.httpOnly = false
             self.secure = false
 
-            for component in components[1...] {
-                switch self.parseComponent(component) {
-                case (nil, nil):
-                    continue
-                case ("path", .some(let value)):
-                    self.path = value
-                case ("domain", .some(let value)):
-                    self.domain = value
-                case ("expires", let value):
-                    guard let value = value else {
-                        continue
-                    }
-
-                    let formatter = DateFormatter()
-                    formatter.locale = Locale(identifier: "en_US")
-                    formatter.timeZone = TimeZone(identifier: "GMT")
+            var parsedPath: String.UTF8View.SubSequence?
+            var parsedDomain: String.UTF8View.SubSequence?
 
-                    formatter.dateFormat = "EEE, dd MMM yyyy HH:mm:ss z"
-                    if let date = formatter.date(from: value) {
-                        self.expires = date
-                        continue
+            for component in components {
+                switch component.parseCookieComponent() {
+                case ("path", let value)?:
+                    // Unlike other values, unspecified and empty paths reset to the default path.
```

Comment:
Can we cite comments like this? I love to see them and I'm really pleased this one is here, but it makes my life much easier as a reviewer (and future reader) if I can rapidly follow the citation (in this case, RFC 6265 § 5.2.4).